streams/examples/src/main/java/org/apache/kafka/streams/examples/pageview/PageViewTypedJob.java

```diff
@@ -124,9 +112,9 @@ public static void main(String[] args) throws Exception {
         serdeProps.put("JsonPOJOClass", RegionCount.class);
         regionCountSerializer.configure(serdeProps, false);
 
-        KStream<String, PageView> views = builder.stream(stringDeserializer, pageViewDeserializer, "streams-pageview-input");
+        KStream<String, PageView> views = builder.stream(Serdes.String(), Serdes.serdeFrom(null, pageViewDeserializer), "streams-pageview-input");
```

Comment:
> The places where not all serdes are needed are in `KStream.to` and `KStream.from` etc, where you only need serializers and deserializers.

I think, if you already have a full serde, there's no harm to pass the full serde to `to` and `from` (which is kinda obvious, still want to mention it).  The question is rather: are there situations where a user would feel bothered to provide "the missing" half of a full serde; i.e. if they must provide, say, a serializer for some "weird" type that a different user stored into Kafka, if they already struggled to write the deserializer.  I could imagine this to be a problem for new users when, for the first time, they must understand how to deserialize a complex data format like Avro or JSON, and then must also spend time on writing the serializer even though they only need the deserializer part ("gee, it took me already an hour to finish the deserializer part, why am I also forced to write a serializer too?  I only need to read it, not write it!")